### PR TITLE
Add client id to client and link

### PR DESCRIPTION
**Pull request by ZejiaJiang**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by ZejiaJiang** [comment]
https://github.com/Azure/azure-sdk-for-java/issues/22981

File: sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java [hunk 1]
```diff
@@ -525,13 +527,16 @@ protected ReactorReceiver createConsumer(String entityPath, Receiver receiver,
...
         sender.setTarget(target);
-
-        final Source source = new Source();
-        sender.setSource(source);
         sender.setSenderSettleMode(SenderSettleMode.UNSETTLED);
 
         if (linkProperties != null && linkProperties.size() > 0) {
             sender.setProperties(linkProperties);
+            String clientId = (String) linkProperties.get(CLIENT_ID);
+            if (clientId != null && !"".equals(clientId)) {
```
Comment (weidongxu-microsoft) [inline sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java:535]: nit, `CoreUtils.isNullOrEmpty`

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java [hunk 2]
```diff
@@ -274,6 +276,18 @@ private String getAndValidateFullyQualifiedNamespace() {
...
 
+    /**
+     * Sets the client id for the Service Bus client.
+     *
+     * @param clientId The customized identifier for Service Bus client.
+     *
+     * @return The updated {@link ServiceBusClientBuilder} object.
+     */
+    public ServiceBusClientBuilder clientId(String clientId) {
+        this.clientId = clientId;
```
Comment (weidongxu-microsoft) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java:287]: might need some validation

File: sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java [hunk 3]
```diff
@@ -525,13 +527,16 @@ protected ReactorReceiver createConsumer(String entityPath, Receiver receiver,
...
+                final Source source = new Source();
+                source.setAddress(clientId);
+                sender.setSource(source);
```
Comment (weidongxu-microsoft) [inline sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java:538]: why this is moved in clientId condition?

**Pull request closed without merging**